catalog/README.md

```diff
@@ -0,0 +1,37 @@
+# Catalog data for Ion test cases
+
+This directory includes Ion catalog data: shared symbol tables and modules that
+are reused across other Ion documents.
+
+Catalog entities are indexed by a name and version, where the name is a non-empty
+string (sequence of Unicode code points) and the version is a positive integer.
+
+Each Ion file in this directory contains one or more valid catalog entities.
+Each entity is represented as an Ion value following the canonical form of
+either:
+  * a shared symbol table: a struct with the first annotation 
+    `$ion_shared_symbol_table`
+  * a shared module: an S-expression with the first annotation 
+    `$ion_shared_module`
+
+Given a _name_ and _version_, implementations locate the relevant entity as 
+follows:
+
+  * First, look for a file path `<name>.ion` and seek the entity therein with
+    the given version.
+  * If no such file exists, seek the entity in the file `catalog.ion`. 
+
+
+## Constraints
+
+The data in this directory must satisfy these constraints:
+
+  * Every Ion file must consist solely of one or more top-level values denoting 
+    shared symbol tables and/or modules.
+  * Each value MUST contain valid name and version metadata per the specified 
+    canonical form; any other data in the entity MAY be absent or erroneous.   
+  * Except for the catch-all file `catalog.ion`, every entity MUST have a name
+    that matches the file name.
```

Comment:
What's the purpose of this requirement? I thought a file would map to a catalog, but with this restriction it seems like only `catalog.ion` is a true catalog; other files would be limited to a single shared table (with multiple revisions allowed).

Reply:
This entire directory represents the catalog for the rest of this package.

IMO at scale this will be unmanageable as a single file, so I worked out a protocol to allow things to be partitioned sensibly.   The catch-all file is here 90% because it will be a bunch of work to implement the multi-file for the semantic model (for a few technical reasons) and I wanted something quick and dirty.

Reply:
Is the assumption that there will never be a test that uses multiple catalogs? Or different tests that use different catalogs? I could imagine this being potentially useful for some unknown symbol cases, but maybe such tests can remain outside the scope of this package.

Reply:
There's only one catalog: this entire directory.  I'm not sure what it would mean for a test to use multiple catalogs; a catalog is just a name->module map.

The catalog data herein includes symbol table `mnop` which has a missing version, and a version 4 has a gap.  So far that has seemed sufficient to test nominal unknown symbol cases, though we could probably use some more complex cases.

Reply:
I agree, I don't think we have a case for multiple catalogs. Catalogs are configured outside the data stream, and the stuff inside the data stream is really only concerned with things that you can get from a catalog.